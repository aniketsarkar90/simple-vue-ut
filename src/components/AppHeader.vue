<template>
    <div>
        <h3>Math Test</h3>
        <p>What is the sum of the two numbers?</p>
        <div class="inline">
            <p>{{ x1 }} + {{ x2 }} =</p> <input v-model="guess"> <button v-on:click="check">Check Answer</button>
        </div>
        <button id='refresh' v-on:click="refresh">Refresh</button>
        <p>{{message}}</p>
    </div>
</template>

<script>
export default {
  name: 'App',
  data() {
    return {
      x1: Math.ceil(Math.random() * 100),
      x2: Math.ceil(Math.random() * 100),
      guess: "",
      message: ""
    }
  },
  methods: {
    check() {
      if (this.x1 + this.x2 === parseInt(this.guess)) {
        this.message = "SUCCESS!"
      } else {
        this.message = "TRY AGAIN"
      }
    },
    refresh() {
      this.x1 = Math.ceil(Math.random() * 100);
      this.x2 = Math.ceil(Math.random() * 100);
      this.message = "REFRESHED"
    }
  }
}
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
.inline * {
  display: inline-block;
}
img {
  height: 350px;
}
</style>