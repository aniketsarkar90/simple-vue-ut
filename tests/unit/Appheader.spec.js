import {mount} from '@vue/test-utils';
import AppHeader from '@/components/AppHeader.vue';

describe('AppHeader', () => {
    it('has data', () => {
        expect(typeof AppHeader.data).toBe('function')
    })
})

describe('Mounted AppHeader', () => {
    const wrapper = mount(AppHeader);
    it('renders correct markup', () => {
        expect(wrapper.html()).toContain('What is the sum of the two numbers?')
    })

    it('test addition, failure', () => {
        wrapper.setData ({ x1:20, x2: 25, guess: "15" })
        const button = wrapper.find('button')
        button.trigger('click')
        //expect(wrapper.vm.message).toBe('TRY AGAIN')
        expect(wrapper.vm.message).toMatchSnapshot()
    })

    it('test addition, success', () => {
        wrapper.setData ({ x1:2, x2: 8, guess: "10" })
        const button = wrapper.find('button')
        button.trigger('click')
        //expect(wrapper.vm.message).toBe('SUCCESS!')
        expect(wrapper.vm.message).toMatchSnapshot()
    })

    it('refresh button', () => {
        const button = wrapper.find('#refresh')
        button.trigger('click')
        //expect(wrapper.vm.message).toBe('REFRESHED!')
        expect(wrapper.vm.message).toMatchSnapshot()
    })
})