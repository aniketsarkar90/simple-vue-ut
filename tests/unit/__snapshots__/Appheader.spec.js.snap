// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted AppHeader refresh button 1`] = `"REFRESHED"`;

exports[`Mounted AppHeader test addition, failure 1`] = `"TRY AGAIN"`;

exports[`Mounted AppHeader test addition, success 1`] = `"SUCCESS!"`;
